tests/test_rechunk.py

```diff
@@ -130,6 +206,80 @@ def test_rechunk_dataset(
     xarray.testing.assert_equal(ds.compute(), dst.compute())
     assert ds.attrs == dst.attrs
 
+@pytest.mark.parametrize("shape", [(100, 50)])
+@pytest.mark.parametrize("source_chunks", [(10, 50), (100, 5)])
+@pytest.mark.parametrize(
+    "target_chunks",
+    [
+        {"x": 20},  # ? Should this rechunk y? Probably not...
+        {"x": 20, "y": 1e5},
+        {"x": 20, "y": -1},
+    ],
+)
+@pytest.mark.parametrize("max_mem", ["10MB"])
+@pytest.mark.parametrize("target_store", ["target.zarr", "mapper.target.zarr"])
+@pytest.mark.parametrize("temp_store", ["temp.zarr", "mapper.temp.zarr"])
+def test_rechunk_dataset_dimchunks(
+    tmp_path,
+    shape,
+    source_chunks,
+    target_chunks,
+    max_mem,
+    target_store,
+    temp_store,
+):
+    if target_store.startswith("mapper"):
+        fsspec = pytest.importorskip("fsspec")
+        target_store = fsspec.get_mapper(str(tmp_path) + target_store)
+        temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
+    else:
+        target_store = str(tmp_path / target_store)
+        temp_store = str(tmp_path / temp_store)
+
+    a = numpy.arange(numpy.prod(shape)).reshape(shape).astype("f4")
+    a[-1] = numpy.nan
+    ds = xarray.Dataset(
+        dict(
+            a=xarray.DataArray(
+                a, dims=["x", "y"], attrs={"a1": 1, "a2": [1, 2, 3], "a3": "x"}
+            ),
+            b=xarray.DataArray(numpy.ones(shape[0]), dims=["x"]),
+            c=xarray.DataArray(numpy.ones(shape[1]), dims=["y"]),
+        ),
+        coords=dict(
+            cx=xarray.DataArray(numpy.ones(shape[0]), dims=["x"]),
+            cy=xarray.DataArray(numpy.ones(shape[1]), dims=["y"]),
+        ),
+        attrs={"a1": 1, "a2": [1, 2, 3], "a3": "x"},
+    )
+    ds = ds.chunk(chunks=dict(zip(["x", "y"], source_chunks)))
+    options = dict(
+        a=dict(
+            compressor=zarr.Blosc(cname="zstd"),
+            dtype="int32",
+            scale_factor=0.1,
+            _FillValue=-9999,
+        )
+    )
+    rechunked = api.rechunk(
+        ds,
+        target_chunks=target_chunks,
+        max_mem=max_mem,
+        target_store=target_store,
+        target_options=options,
+        temp_store=temp_store,
+    )
+    assert isinstance(rechunked, api.Rechunked)
+    with dask.config.set(scheduler="single-threaded"):
+        rechunked.execute()
+
+    assert dst.a.data.chunksize == target_chunks_expected
```

Comment:
Looks like you have not defined `dst`. In the other test, it is defined as

```python
dst = xarray.open_zarr(target_store, decode_cf=True)
```

Reply:
Well this is embarrassing. Sorry for this fail...Ill fix that right now.